pkg/auth/auth_handler.go

```diff
@@ -185,34 +190,50 @@ func (a *AuthHandler) AuthUserAuth(token string) (interface{}, error) {
 		return nil, errors.Errorf("Missing username in token")
 	}
 
-	err = a.storeAdminInPayload(payload)
-	if err != nil {
-		a.log.Errorf("Unable to fetch user's capabilities: %v", err)
-		return nil, common.ApiErrorWithDefaultInfraError(err, http.StatusUnauthorized)
+	if err := a.storeRoleInUserAuthPayload(payload); err != nil {
+		return nil, err
 	}
 
 	return payload, nil
 }
 
-func (a *AuthHandler) storeAdminInPayload(payload *ocm.AuthPayload) error {
+func (a AuthHandler) storeRoleInUserAuthPayload(payload *ocm.AuthPayload) error {
 	payloadKey := payload.Username + "_is_admin"
-	payloadFromCache, found := a.client.Cache.Get(payloadKey)
-	if found {
-		payload.IsAdmin = payloadFromCache.(*ocm.AuthPayload).IsAdmin
-		return nil
+	if payloadFromCache, existInCache := a.client.Cache.Get(payloadKey); existInCache {
+		payload.Role = payloadFromCache.(*ocm.AuthPayload).Role
```

Comment:
Cache handling to not be here, but in AuthUserAuth()

Also you need to add to the cache if user it was not in cache and error < 500